sqlmesh/migrations/v0013_serde_using_model_dialects.py

```diff
@@ -0,0 +1,60 @@
+"""Serialize SQL using the dialect of each model."""
+import json
+
+import pandas as pd
+from sqlglot import exp, transpile
+
+from sqlmesh.utils.jinja import has_jinja
+
+
+def migrate(state_sync):  # type: ignore
+    engine_adapter = state_sync.engine_adapter
+    schema = state_sync.schema
+    snapshots_table = f"{schema}._snapshots"
+
+    new_snapshots = []
+
+    for name, identifier, version, snapshot, kind_name in engine_adapter.fetchall(
+        exp.select("name", "identifier", "version", "snapshot", "kind_name").from_(snapshots_table)
+    ):
+        parsed_snapshot = json.loads(snapshot)
+        model = parsed_snapshot["model"]
+        dialect = model["dialect"]
+
+        # Read using the SQLGlot dialect, write using the model's dialect
+        if "query" in model and not has_jinja(model["query"]):
+            model["query"] = transpile(model["query"], read="", write=dialect, identity=False)[0]
+
+        for statement_kind in ("pre_statements_", "post_statements_"):
```

Comment:
Makes sense, updated & fixed another bug I found in my implementation. I was calling `has_jinja` on the `model["pre_statements"]` list instead of each actual statement.